### DISCUSSION TITLE
Internalization of panel plugins

### DISCUSSION THREAD

**Discussion by andreasgerstmayr** [perses/perses#2287] [Ideas]
## Problem statement
Some panels contain hardcoded static English text, for example

TraceTable header:
![image](https://github.com/user-attachments/assets/2ac227b4-fe32-434e-8055-2b3e65139eff)

or parts of the Gantt chart panel:
![image](https://github.com/user-attachments/assets/4b6de7dd-aa54-4149-981e-e75ce9996949)

When embedding these panels in pages which are localized (e.g. OpenShift Console), we'd like to translate all (static) text.

Many translation libraries work by running the default language or a specific key through a translation function, which then returns the translated string. For example: https://react.i18next.com/guides/quick-start#using-the-hook

## Suggested solution
Provide a translation function `(text: string) => string` either as a prop or React Provider, which the panel plugins can use for all translatable texts. Downstream applications which embed the panel can overwrite this provider / prop.

### Where should the translation strings for every language be stored?
* next to the (dynamic) plugin
* at the parent level (Perses UI or some downstream app, e.g. OpenShift Console)?

Note that downstream products may want to support more languages than we want to maintain in Perses.

## Related issues
Slightly related: https://github.com/perses/perses/discussions/2112. The current issue is about translating UI texts in panel plugins (and potentially also the Perses UI, e.g. the navigation etc.), while the linked issue is about translating static texts of dashboards (e.g. panel titles etc.)

**Comment by Nexucis** [comment]
We didn't discuss since a while, but a message in #2112 revived me this topic.

On my side I won't be opposed if someone would like to do it and provide an easy to use/implement solution. On Amadeus side we don't have the bandwidth to tackle this topic currently, but happy to support any work in this area
